content/topics/file-and-directory-naming/javascript/_index.md

```diff
@@ -0,0 +1,67 @@
+---
+content_type: topic
+prerequisites:
+  hard:
+    - topics/file-and-directory-naming/general 
+
+ready: true
+tags:
+  - File and directory naming
+title: File and directory naming in JavaScript
+---
+
+## How should I name my JavaScript files and directories?
+
+When thinking about how to name files in JavaScript, think of the file as an entity within the code. This means, if you have a file with a class called `Calculator`, then it also makes sense to name the file `Calculator.js` ; notice that the naming convention utilizes _PascalCase_. The same thing should be done if a file contains a function - if you have a file containing a single function called `calculateDateOfBirth`, it should live inside a _camelCased_ file called `calculateDateOfBirth.js` . The same logic should apply for naming directories that live inside your **src** directory. Remember to be always consistent.
```

Comment:
in React it's common practice to capitalise your file names if they contain components. Eg `MyComponent`.  It's a way to say that the file contains something very special. 

Most classes aren't that special. And it's quite normal to import classes and functions from the same place. It's better to be consistent. Use the same naming convention for all js files.

It would also be bad to recommend one naming convention here and then insist on a different one in all our actual projects. 

What convention to we use?